error-prone-contrib/src/test/java/tech/picnic/errorprone/bugpatterns/util/MoreMatchersTest.java

```diff
@@ -47,9 +50,33 @@ void hasMetaAnnotation() {
         .doTest();
   }
 
-  /** A {@link BugChecker} that delegates to {@link MoreMatchers#hasMetaAnnotation(String)} . */
+  @Test
+  void hasLombokDataAnnotation() {
+    CompilationTestHelper.newInstance(LombokDataAnnotationMatcher.class, getClass())
+        .addSourceLines("Data.java", "package lombok;", "", "public @interface Data {}")
+        .addSourceLines(
+            "A.java",
+            "import lombok.Data;",
+            "",
+            "@Data",
+            "// BUG: Diagnostic contains:",
+            "public class A {",
+            "  private String field;",
+            "",
+            "  static class B {}",
+            "",
+            "  @Data",
+            "  // BUG: Diagnostic contains:",
+            "  static class C {}",
+            "}")
+        .addSourceLines("D.java", "public class D {}")
+        .doTest();
+  }
+
+  /** A {@link BugChecker} that delegates to {@link MoreMatchers#hasMetaAnnotation(String)}. */
   @BugPattern(summary = "Interacts with `MoreMatchers` for testing purposes", severity = ERROR)
-  public static final class TestMatcher extends BugChecker implements AnnotationTreeMatcher {
+  public static final class TestTemplateMatcher extends BugChecker
+      implements AnnotationTreeMatcher {
```

Comment:
In #642 I renamed this type to `HasMetaAnnotationTestChecker`, in line with approaches taken elsewhere.